lib/Service/CollectiveShareService.php

```diff
@@ -156,32 +197,46 @@ private function isShareEditable(IShare $folderShare): bool {
 	/**
 	 * @param string         $userId
 	 * @param CollectiveInfo $collective
+	 * @param PageInfo|null  $pageInfo
 	 *
 	 * @return CollectiveShareInfo
 	 * @throws NotFoundException
 	 * @throws NotPermittedException
 	 */
-	public function createShare(string $userId, CollectiveInfo $collective): CollectiveShareInfo {
+	public function createShare(string $userId, CollectiveInfo $collective, ?PageInfo $pageInfo): CollectiveShareInfo {
 		if (!$collective->canShare()) {
 			throw new NotPermittedException($this->l10n->t('You are not allowed to share %s', $collective->getName()));
 		}
 
-		if (null !== $this->findShare($userId, $collective->getId())) {
+		$pageId = 0;
+		$nodeId = 0;
+		if ($pageInfo) {
+			$pageId = $pageInfo->getId();
+			$file = $this->pageService->getPageFile($collective->getId(), $pageId, $userId);
+			if (NodeHelper::isIndexPage($file)) {
+				$nodeId = $file->getParent()->getId();
+			} else {
+				$nodeId = $pageInfo->getId();
+			}
+		}
+
+		if (null !== $this->findShare($userId, $collective->getId(), $nodeId)) {
 			throw new NotPermittedException($this->l10n->t('A share for collective %s exists already', $collective->getName()));
```

Comment:
> A share for collective %s exists already

or the particular page in question, right?

Reply:
Same here. We want to support several shares now, so let's just get rid of this check.